### PR TITLE
Adds better error handling to kaggle_secrets.py and calls UserSecrets…

### PR DESCRIPTION
**Pull request by vimota**
…Client from KaggleKernelCredentials

### CODE REVIEW

File: patches/kaggle_gcp.py [hunk 1]
```diff
@@ -46,4 +48,4 @@ def __init__(self):
             project=data_proxy_project, credentials=anon_credentials
         )
         # TODO: Remove this once https://github.com/googleapis/google-cloud-python/issues/7122 is implemented.
-        self._connection = _DataProxyConnection(self)
+        self._connection = _DataProxyConnection(self)
```
Comment (rosbo) [inline patches/kaggle_gcp.py]: Add back the newline at the end of file. Ditto for the other files.

File: patches/kaggle_secrets.py [hunk 2]
```diff
@@ -12,7 +10,7 @@
 
 _KAGGLE_DEFAULT_URL_BASE = "https://www.kaggle.com"
 _KAGGLE_URL_BASE_ENV_VAR_NAME = "KAGGLE_URL_BASE"
-_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME = "KAGGLE_USER_SECRETS_TOKEN_KEY"
+_KAGGLE_USER_SECRETS_TOKEN_ENV_VAR_NAME = "KAGGLE_USER_SECRETS_TOKEN"
```
Comment (rosbo) [inline patches/kaggle_secrets.py]: What's the plan for this env var rename?

We set it at worker startup here: https://github.com/Kaggle/Kaggle.SDS.Worker/blob/ci/kernels/worker/init/setup.sh#L147

Comment (vimota) [reply patches/kaggle_secrets.py]: Right, I had made a mistake originally calling TOKEN_KEY here, what we really want is the TOKEN itself: https://github.com/Kaggle/Kaggle.SDS.Worker/blob/7e5c918306f1424739dcb104b39e2449980320fd/kernels/worker/src/kernel_run_container_commands.py#L65. The token key is used to generate the token in the user_secrets_token_gen.py

File: patches/kaggle_gcp.py (hunk 1 above)
Comment (vimota) [reply patches/kaggle_gcp.py]: Done!

**Pull request merged**